src/frequenz/client/reporting/_client.py

```diff
@@ -244,48 +245,15 @@ def dt2ts(dt: datetime) -> PBTimestamp:
         try:
             stream = cast(
                 AsyncIterator[PBReceiveMicrogridComponentsDataStreamResponse],
-                self._stub.ReceiveMicrogridComponentsDataStream(
+                self.stub.ReceiveMicrogridComponentsDataStream(
                     request, metadata=self._metadata
                 ),
             )
-            # grpc.aio is missing types and mypy thinks this is not
-            # async iterable, but it is.
             async for response in stream:
                 if not response:
                     break
-
                 yield ComponentsDataBatch(response)
```

Comment:
At some point we might want to consider to provide a way to wrap these streaming methods too in `client-base`, so we don't need to have these casts everywhere, which are also error prone.